src/java/org/apache/cassandra/cache/InstrumentingCache.java

```diff
@@ -84,6 +86,11 @@ public void setCapacity(long capacity)
         map.setCapacity(capacity);
     }
 
+    public boolean enabled()
+    {
+        return enabled;
```

Comment:
if we call "setCapacity" from CacheService as shown in other PR, then this should be "return map.capacity() > 0" to dynamically reflect if cache is enabled or not.

Reply:
hah well ... but we can not do that, can we, because every single time we should ask for "capacity" again which locks ...